analyzers/slice_bounds.go

```diff
@@ -118,32 +118,41 @@ func runSliceBounds(pass *analysis.Pass) (interface{}, error) {
 			if i == 1 {
 				bound = invBound(bound)
 			}
-			for _, instr := range block.Instrs {
-				if _, ok := issues[instr]; ok {
-					switch bound {
-					case lowerUnbounded:
-						break
-					case upperUnbounded, unbounded:
-						delete(issues, instr)
-					case upperBounded:
-						switch tinstr := instr.(type) {
-						case *ssa.Slice:
-							lower, upper := extractSliceBounds(tinstr)
-							if isSliceInsideBounds(0, value, lower, upper) {
-								delete(issues, instr)
-							}
-						case *ssa.IndexAddr:
-							indexValue, err := extractIntValue(tinstr.Index.String())
-							if err != nil {
-								break
-							}
-							if isSliceIndexInsideBounds(0, value, indexValue) {
-								delete(issues, instr)
+			var processBlock func(block *ssa.BasicBlock)
```

Comment:
nit: I would use the short variable declaration here.

Reply:
You mean `processBlock := func(...)`? That's what I tried doing at first but the compiler yells since we're calling it recursively; I assume it has to see the declaration in its scope to use it.